symengine/printer.cpp

```diff
@@ -382,7 +382,7 @@ void StrPrinter::bvisit(const Add &x)
     for (const auto &p : dict) {
         std::string t;
         if (eq(*(p.second), *one)) {
-            t = this->apply(p.first);
+            t = parenthesizeLT(p.first, PrecedenceEnum::Mul);
```

Comment:
I think we should allow a precedence level that is lower than `Add` for Relationals and make this `Add`.
`1 + Add` shouldn't parenthesize `Add`, but since symengine doesn't keep `x + (y+z)` as it is and flattens to `x+y+z`, what you do here is okay, but it's better to have a new level so that it won't break in the future.